### ISSUE TITLE
Avoid clearing environment if pip is not able to install new requirements

### ISSUE THREAD

**Issue by rasmusoej** [SallingGroup-AI-and-ML/venv-cli#40]
If you run `venv install -r somefile.txt` but `somefile.txt` does not exist then the environment is first cleared and then `pip` gives an error (see attachment). It could be a nice feature to first check if the file exists before clearing the environment. The same problem occurs if you try to install a package that is not known to `pip`.

Of course it is bad practice to try to install something that does not exist and in most cases you are able to recreate the environment which was cleared, but it would be nice to not lose the entire environment due to a spelling mistake for instance ;) 

![Screenshot from 2024-03-19 07-25-03](https://github.com/SallingGroup-AI-and-ML/venv-cli/assets/124035247/348ce931-a73c-4a99-9917-91e8c0f429ba)

**Comment by connesy** [comment]
@rasmusoej We can check for the requirements file earlier, but I don't see an easy way to ensure that the specified packages are known to `pip` beforehand. We only know that once we try to install them.

Also, from a user perspective, if you're trying to install a package and misspell it (thus clearing the environment without installing anything), I would assume the next logical step would be to try again with the correct spelling, in which case you would reinstall the environment anyway.

Did you encounter a different use case where this is not the case?

**Comment by rasmusoej** [comment]
No, I agree that there is no good way to check for individual packages, but I think checking the file earlier would be a cool addition. 

The only other problem I can think is if you try to install a package that does not exist and the environment is cleared before `pip` throws the error, the package is still added to `requirements.txt` file. So when you try to recreate the environment by `venv install -r requirements.txt` (or try to run `venv install` with the correct package name) you get the same error again because `pip` still tries to install the non-existent package. The solution is to manually remove it from `requirements,txt`, and that might be okay :)

**Comment by connesy** [comment]
Actually, `venv install` creates a backup of `requirements.txt` when it begins. If everything runs successfully, this is deleted afterwards.

We could add a "rollback" where, if the `pip install`-step fails, we restore the `requirements.txt` file from the backup automatically /right now you can do it manually by `mv requirements.txt.bak requirements.txt`.

**Comment by connesy** [comment]
Example of trying to install a non-existing package and manually restoring the backup `requirements.txt` file:
![image](https://github.com/SallingGroup-AI-and-ML/venv-cli/assets/13164166/61984b6f-7728-49f9-85b3-dc3bbb5065c8)
